dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityPreviewServlet.java

```diff
@@ -0,0 +1,359 @@
+            // Our designed template
+            template = VelocityUtil.getEngine()
+                .getTemplate((String) dotThemeData.get("templatePath"));
+        } else {
+            template = VelocityUtil.getEngine()
+                .getTemplate(PageMode.PREVIEW_MODE.name() + File.separator + cmsTemplate.getIdentifier() + "."
+                        + VelocityType.TEMPLATE.fileExtension);
+        }
+
+
+
+        PrintWriter out = response.getWriter();
+        request.setAttribute("velocityContext", context);
+        try {
+
+            if (builder.getWidgetPreExecute() != null) {
+                VelocityUtil.getEngine()
+                    .evaluate(context, out, "", builder.getWidgetPreExecute());
+            }
+            if (hostVariablesTemplate != null)
+                hostVariablesTemplate.merge(context, out);
+            template.merge(context, out);
+
+        } catch (ParseErrorException e) {
+            out.append(e.getMessage());
+        }
+    }
+
+
+    protected void doEditMode(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        PageMode mode = PageMode.PREVIEW_MODE;
+        String uri = request.getRequestURI();
+
+        Host host = hostWebAPI.getCurrentHost(request);
+
+
+        // Getting the user to check the permissions
+        User user = com.liferay.portal.util.PortalUtil.getUser(request);
+
+
+        // Getting the identifier from the uri
+        Identifier id = APILocator.getIdentifierAPI()
+            .find(host, uri);
+        request.setAttribute("idInode", String.valueOf(id.getId()));
+        Logger.debug(VelocityPreviewServlet.class, "VELOCITY HTML INODE=" + id.getId());
+
+        Template template;
+        Template hostVariablesTemplate = null;
+
+        // creates the context where to place the variables
+        response.setContentType(CHARSET);
+        request.setAttribute("EDIT_MODE", Boolean.TRUE);
+        Context context = VelocityUtil.getWebContext(request, response);
+
+        IHTMLPage htmlPage = VelocityUtil.getPage(id, request, false, context);
+
+        context.put("dotPageContent", new ContentMap(((Contentlet) htmlPage), user, mode, host, context));
+
+
+        com.dotmarketing.portlets.templates.model.Template cmsTemplate = APILocator.getHTMLPageAssetAPI()
+            .getTemplate(htmlPage, true);
+
+        if (cmsTemplate == null) {// DOTCMS-4051
+            cmsTemplate = new com.dotmarketing.portlets.templates.model.Template();
+            Logger.debug(VelocityPreviewServlet.class, "HTMLPAGE TEMPLATE NOT FOUND");
+        }
+
+        Identifier templateIdentifier = APILocator.getIdentifierAPI()
+            .find(cmsTemplate);
+
+        Logger.debug(VelocityPreviewServlet.class, "VELOCITY TEMPLATE INODE=" + cmsTemplate.getInode());
+
+        VelocityUtil.makeBackendContext(context, htmlPage, cmsTemplate.getInode(), id.getURI(), request, mode, host);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "makeBackendContext". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)